app/models/order.rb

```diff
@@ -1,10 +1,33 @@
-class Order <ApplicationRecord
+class Order < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
 
   has_many :item_orders
   has_many :items, through: :item_orders
+  belongs_to :user
+
+  enum status:  %w(pending packaged shipped cancelled)
 
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def total_quantity
+    item_orders.sum(:quantity)
+  end
+
+  def status_check
+    if all_fulfilled? && self.status != "packaged"
+      self.status = "packaged"
+      self.save
+      self.status
+    else
+      self.status
+    end
+  end
+
+  def all_fulfilled?
```

Comment:
I agree Shaunda, we can come back and refactor this to be AR once we are done with the main stories. Been thinking about it all day in the back of my head.